scripts/globals/instance.lua

```diff
@@ -414,20 +428,49 @@ xi.instance.onInstanceCreatedCallback = function(player, instance)
     -- If you're in the official entrance zone, try and playout the
     -- entrance animation. Otherwise: go straight to the instance
     if player:getZoneID() == instance:getEntranceZoneID() then
+        -- join initiating player as commander
+        player:setInstance(instance)
+
         -- This packet will trigger the end of the blocking
         -- cutscene and xi.instance.onEventFinish will handle
         -- the transportation
         for _, v in ipairs(player:getParty()) do
-            if v:getID() ~= player:getID() then
-                v:startEvent(unpack(lookupEntry[4]))
-            end
+            if v:getZoneID() == player:getZoneID() then
+                if v:getID() ~= player:getID() then
+                    -- player will be brought into instance either way
+                    -- this makes the animation trigger reliably
+                    v:release()
+                    v:startEvent(unpack(lookupEntry[4]))
+                end
 
-            v:setInstance(instance)
-            local npc = player:getEventTarget()
-            if npc ~= nil then
-                v:instanceEntry(npc, 4)
+                v:setInstance(instance)
+                local npc = player:getEventTarget()
+                if npc ~= nil then
+                    v:instanceEntry(npc, 4)
+                end
+
+                v:timer(35000, function(playerArg)
+                    -- failsafe to bring all players into instance
+                    -- taking care not to shadow any existing variables
+                    -- if a player doesn't receive the event packet, the whole party will be stuck in blackscreen
+                    -- timer is destroyed if onEventFinish works properly and player gets zoned
+                    -- a properly-functioning event loop will take 20s to zoning into the instance
+                    -- this _should_ be completely unnecessary due to the looping logic with INSTANCE_REQUESTED, but just in case
+                    local instanceArg = playerArg:getInstance()
+                    if instanceArg then
+                        print(fmt('Player {} failed to cleanly transition into instance event, forcing entry via setPos.', playerArg:getName()))
```

Comment:
Like you mentioned in your comment, `fmt` needs a global exception - you can add it as a commit in the PR if you like